include/fiction/utils/math_utils.hpp

```diff
@@ -49,6 +50,79 @@ T integral_abs(const T n) noexcept
 
     return static_cast<T>(std::abs(static_cast<int64_t>(n)));  // needed to solve ambiguity of std::abs
 }
+/**
+ * Calculates the binomial coefficient C(n, k).
+ *
+ * @param n The total number of items.
+ * @param k The number of items to choose from n.
+ * @return The binomial coefficient C(n, k).
+ */
+inline uint64_t binomial_coefficient(uint64_t n, uint64_t k) noexcept
+{
+    if (k > n)
+    {
+        return 0;
+    }
+    uint64_t result = 1;
+    if (2 * k > n)
+    {
+        k = n - k;
+    }
+    for (uint64_t i = 1; i <= k; i++)
+    {
+        result = result * (n + 1 - i) / i;
+    }
+    return result;
+}
+/**
+ * This function generates all combinations of `k` unique unsigned integers within the specified range [0, n-1].
+ *
+ * @param combinations Vector to store the generated combinations.
+ * @param current_combination Current combination being constructed.
+ * @param total_elements The total number of elements (maximum value).
+ * @param current_index Current index in the iteration.
+ * @param remaining_elements Number of elements remaining to be added to the combination.
+ */
+inline void generate_unsigned_integer_combinations_helper(std::vector<std::vector<uint64_t>>& combinations,
+                                                          std::vector<uint64_t>&              current_combination,
+                                                          const uint64_t total_elements, const uint64_t current_index,
+                                                          const uint64_t remaining_elements) noexcept
+{
+    // If we have selected 'k' elements, add the current combination to the result.
+    if (remaining_elements == 0)
+    {
+        combinations.push_back(current_combination);
+        return;
+    }
+
+    // Iterate through the remaining elements.
+    for (uint64_t i = current_index; i <= total_elements - remaining_elements; ++i)
+    {
+        current_combination.push_back(i);
+        generate_unsigned_integer_combinations_helper(combinations, current_combination, total_elements, i + 1,
+                                                      remaining_elements - 1);
+        current_combination.pop_back();  // Remove the last added element for backtracking.
+    }
+}
```

Comment:
We use the `combinations` library for use cases like these. Please see https://howardhinnant.github.io/combinations/combinations.html for the documentation.

Reply:
@marcelwa I am not sure if there is a function that does the same thing.

Reply:
@marcelwa Maybe I didn't mention it clearly enough, but in this case the order of the entries is negligible.

Reply:
Correct me if I'm wrong but I think this is exactly what you want:

```c++
std::vector<int> numbers(n);
std::iota(numbers.begin(), numbers.end(), 0);

combinations::for_each_combination(numbers.begin(), numbers.begin() + k, numbers.end(),
                                   [](const auto begin, [[maybe_unused]] const auto end)
                                   {
                                       for (auto i = 0u; i < k; ++i)
                                       {
                                           std::cout << *(begin + i) << " ";
                                       }
                                       std::cout << "\n";

                                       return false;  // keep looping
                                   });
```

Reply:
@marcelwa Ah, great! Thanks.